stdlib/fun.ml

```diff
@@ -20,6 +20,12 @@ let negate p v = not (p v)
 
 exception Finally_raised of exn
 
+let _ = Printexc.register_printer (function
+  | Finally_raised exn -> Some ("re-raised by Fun.protect: "
```

Comment:
I don't understand why you say `re-raised` ? It's simply the `finally` that raised. Somehow I find the current error message based on the exn name more clear !

Reply:
I removed re-raised now. This was a reference to the stack trace which says "re-raised from ..." at the location where Fun.protect raises `Finally_raised`.